src/Raven.Server/Web/Studio/Processors/BucketsHandlerProcessorForGetBuckets.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Util;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Sharding;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Web.Studio.Processors
+{
+    internal class BucketsHandlerProcessorForGetBuckets : AbstractBucketsHandlerProcessorForGetBuckets<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public BucketsHandlerProcessorForGetBuckets([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+        
+        protected override ValueTask<BucketsResults> GetBucketsResults(DocumentsOperationContext context, int fromBucket, int toBucket, int range, CancellationToken token)
+        {
+            if (ShardHelper.TryGetShardNumberAndDatabaseName(RequestHandler.DatabaseName, out string shardedDatabaseName, out int shardNumber) == false)
+            {
+                throw new InvalidOperationException();
+            }
+
+            using (context.OpenReadTransaction())
+            {
+                long shardBucketsSize = 0;
+                var ranges = new Dictionary<int, BucketRange>();
+                foreach (var bucket in ShardedDocumentsStorage.GetBucketStatistics(context, fromBucket, toBucket))
+                {
+                    var id = (bucket.Bucket / range) * range;
```

Comment:
No, we need that to be the _start_ of the range, so we basically just align that to the range default.
This is important, since we merge results from multiple shards to get the values. And it gives better output in the JSON for manual review.